platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -299,9 +286,7 @@ void OfflineDownload::ensureResource(const Resource& resource,
                 return;
             }
 
-            if (status.complete()) {
-                setState(OfflineRegionDownloadState::Inactive);
-            }
+            continueDownload();
```

Comment:
A request that errors will be retried after some delay. So in that sense it's still "active" and consuming resources, notably the request object, its timer, and network resources when the timer fires.

We _could_ try to squeeze in subsequent requests while we wait for the errored request to retry. But that risks overloading the upstream request queue -- defeating our own metering here -- if there are a lot of errored requests that all come up for retry at the same time.

Many times, the cause of a request error will apply to many requests of the same type. For instance if a server is unreachable, all the requests to that host are going to error. In that case, continuing to try subsequent resources after the first few errors is fruitless anyway.

I'll add the above rationale as a comment.